packages/core/src/components/control-bar/control-bar.ts

```diff
@@ -496,6 +602,72 @@ export class ProtspaceControlBar extends LitElement {
     }
   }
 
+  private _handleDataSplit(event: Event) {
+    const customEvent = event as CustomEvent;
+    const { splitHistory, splitMode } = customEvent.detail;
+    this.splitHistory = splitHistory;
+    this.splitMode = splitMode;
+    this.selectedProteinsCount = 0;
+    this.requestUpdate();
+  }
+
+  private _handleDataSplitReset(event: Event) {
+    const customEvent = event as CustomEvent;
+    const { splitHistory, splitMode } = customEvent.detail;
+    this.splitHistory = splitHistory;
+    this.splitMode = splitMode;
+    this.selectedProteinsCount = 0;
+    
+    // Re-enable selection when split is reset (back to full data)
+    this._selectionDisabled = false;
+    
+    this.requestUpdate();
+  }
+
+  private _handleAutoDisableSelection(event: Event) {
```

Comment:
This merges logic plus notification. Please instead create an event or somthing to be handled by the actual interface. 

I would usggest completely implementing a notification handler and the actual notification moves into the utils as optional addon. Most frameworks can do it nicer anyway.